compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -147,10 +147,12 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         let mut expected = source.descr_expected();
         let path_str = Segment::names_to_string(path);
         let item_str = path.last().unwrap().ident;
-        let (base_msg, fallback_label, base_span, could_be_expr) = if let Some(res) = res {
+        let (base_msg, fallback_label, base_span, could_be_expr, suggestion) = if let Some(res) =
```

Comment:
i dislike large tuples like this. it's just personal preference, but what's your take on moving these 4 fields to a helper struct?

Reply:
I agree with you. I'll fix it just like you said.